src/listener.ts

```diff
@@ -8,36 +8,182 @@ import { writeFromReadableStream } from './utils'
+  },
+  get ok() {
+    return this.getResponseCache().ok
+  },
+  get redirected() {
+    return this.getResponseCache().redirected
+  },
+  get statusText() {
+    return this.getResponseCache().statusText
+  },
+  get trailers() {
+    return this.getResponseCache().trailers
+  },
+  get type() {
+    return this.getResponseCache().type
+  },
+  get url() {
+    return this.getResponseCache().url
+  },
+  arrayBuffer() {
+    return this.getResponseCache().arrayBuffer()
+  },
+  blob() {
+    return this.getResponseCache().blob()
+  },
+  clone() {
+    return this.getResponseCache().clone()
+  },
+  error() {
+    return this.getResponseCache().error()
+  },
+  formData() {
+    return this.getResponseCache().formData()
+  },
+  json() {
+    return this.getResponseCache().json()
+  },
+  redirect() {
+    return this.getResponseCache().redirect()
+  },
+  text() {
+    return this.getResponseCache().text()
+  },
+}
 
-    const headerRecord: [string, string][] = []
-    const len = incoming.rawHeaders.length
-    for (let i = 0; i < len; i += 2) {
-      headerRecord.push([incoming.rawHeaders[i], incoming.rawHeaders[i + 1]])
-    }
+function newResponse(body: BodyInit | null, init?: ResponseInit): Response {
+  const res = {
+    body,
+    init,
+    status: init?.status || 200,
+    __cache:
+      typeof body === 'string'
+        ? [body, (init?.headers || {}) as Record<string, string>]
+        : undefined,
+  } as unknown as Response
+  Object.setPrototypeOf(res, responsePrototype)
+  return res
+}
+Object.defineProperty(global, 'Response', {
+  value: newResponse,
+})
 
-    const init = {
-      method: method,
-      headers: headerRecord,
-    } as RequestInit
+function newRequestFromIncoming(
+  method: string,
+  url: string,
+  incoming: IncomingMessage | Http2ServerRequest
+): Request {
+  const headerRecord: [string, string][] = []
+  const len = incoming.rawHeaders.length
+  for (let i = 0; i < len; i += 2) {
+    headerRecord.push([incoming.rawHeaders[i], incoming.rawHeaders[i + 1]])
+  }
 
-    if (!(method === 'GET' || method === 'HEAD')) {
-      // lazy-consume request body
-      init.body = Readable.toWeb(incoming) as ReadableStream<Uint8Array>
-      // node 18 fetch needs half duplex mode when request body is stream
-      ;(init as any).duplex = 'half'
-    }
+  const init = {
+    method: method,
+    headers: headerRecord,
+  } as RequestInit
+
+  if (!(method === 'GET' || method === 'HEAD')) {
+    // lazy-consume request body
+    init.body = Readable.toWeb(incoming) as ReadableStream<Uint8Array>
+    // node 18 fetch needs half duplex mode when request body is stream
+    ;(init as any).duplex = 'half'
+  }
+
+  return new Request(url, init)
+}
 
-    let res: Response
+const requestPrototype: Record<string, any> = {
+  getRequestCache() {
+    return (this.requestCache ||= newRequestFromIncoming(this.method, this.url, this.incoming))
+  },
+  get body() {
+    return this.getRequestCache().body
+  },
```

Comment:
This would make a huge difference for devs migrating from Express to Hono in the next few weeks. 🚀 

Would it be possible to also support a built-in middleware such as body-parser (or a how-to guide) via config? cc @yusukebe 

Thanks!

Reply:
Hi @rafaell-lycan!
Thanks for your comment.

Do you think this will cause problems? If so, please let us know what specific issues you see.

> This would make a huge difference for devs migrating from Express to Hono in the next few weeks. 🚀